nbhttp/websocket/upgrader.go

```diff
@@ -66,7 +67,10 @@ func NewUpgrader() *Upgrader {
 			u.conn.Close()
 			return
 		}
-		c.WriteMessage(PongMessage, []byte(data))
+		err := c.WriteMessage(PongMessage, []byte(data))
+		if err != nil {
+			u.Close(nil, err)
+		}

```

Comment:
If c.WriteMessage failed, the nbio.Conn will auto Close, u.Close does not close the Conn.
What we should do maybe is  to avoid the compressor fail:
```golang
if compress {
	compress = true
	w := &writeBuffer{
		Buffer: bytes.NewBuffer(mempool.Malloc(len(data))),
	}
	defer w.Close()
	w.Reset()
	cw := compressWriter(w, c.compressionLevel)
	_, err := cw.Write(data)
	if err != nil {
		return err // avoid this err
	}
	cw.Close()
	data = w.Bytes()
}
```